### ISSUE TITLE
ui-select 0.17.1 repeat error

### ISSUE THREAD

**Issue by Blaze34** [formly-js/angular-formly#681]
Yesterday ui-select was updated and angular-formly stop working.

This is [jsbin example](http://jsbin.com/vivojivayu/1/edit?html,output)

**Comment by thachp** [comment]
I am experiencing similar issue.  I'm getting this error...

    Error: [ui.select:iexp] Expected expression in form of '_item_ in _collection_[ track by _id_]' but got '{{to.ngOptions}}'.

**Comment by Luddinus** [comment]
same issue here

It works if you add 'option[to.valueProp] as option in to.options | filter: $select.search' instead of {{to.ngOptions}}

**Comment by immortalcy** [comment]
It would be nice if we could have a more generic solution instead of changing the template to only support that, since that means that in any case where you would like to have other data-repeat value you wouldn't be able to. You would be 'forced' to create a new template and that is something angular-formly doesn't intent to do.

**Comment by kentcdodds** [comment]
@immortalcy, actually, if you look at the implementation of the example above, it allows you to specify the `ngOptions` at the field level, and actually you can specify the `ngOptions` attribute for a specific field using `ngModelAttrs` as well. I think that's the generic solution you're looking for...

**Comment by immortalcy** [comment]
Yes but with Luddinus solution you won't be able to use to.ngOptions, am I missing something here?

**Comment by jose-brito-ptc** [comment]
I'm also experiencing the same issue. I tried @Luddinus workaround, which works, but that goes against  the purpose of using custom types with formly.

Besides that, I have specified the group-by property, which also stopped working. Here's my code:

`<ui-select-choices data-repeat="{{to.ngOptions}}" group-by="{{to.groupByProp}}">`

By using @Luddinus workaround, I get the following error in group-by:

`Error: [$parse:syntax] Syntax Error: Token '{' invalid key at column 2 of the expression [{{to.groupByProp}}] starting at [{to.groupByProp}}].`

Any suggestions for both of these issues?

**Comment by vmartinetti** [comment]
+1 same problem and same workaround of @Luddinus

**Comment by mohanskumar** [comment]
+1 Same issue here, I need to dynamically build the repeat string and bind it. falling back to 0.17.0

**Comment by blowsie** [comment]
@Luddinus's solution doesn't work for me with the latest version. Its also worth considering that you might want to apply other filters like orderBy

**Comment by arickampschaefer** [comment]
+1 same issue, and @Luddinus's solution doesn't work for me, so I've also fallen back to 0.17.0

**Comment by hustshawn** [comment]
@Luddinus Works well. 'Formly' officially correct this here?
 http://angular-formly.com/#/example/integrations/ui-select-angular-1-4

**Comment by oscar-b** [comment]
Also see https://github.com/angular-ui/ui-select/issues/1640

**Comment by JR-Utily** [comment]
@hustshawn 
no, there is no correction, you can see in the html they included v0.16 of ui-select and the regression was introduced in v0.17.1
`<script src="https://cdn.rawgit.com/angular-ui/ui-select/v0.16.0/dist/select.js"></script>`

**Comment by blowsie** [comment]
https://github.com/formly-js/angular-formly/issues/441
